app/modules/fileuploads/resources.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=bad-continuation
+"""
+RESTful API FileUploads resources
+--------------------------
+"""
+
+import logging
+
+from flask import send_file
+from flask_login import current_user  # NOQA
+from flask_restx_patched import Resource
+from flask_restx_patched._http import HTTPStatus
+
+from app.extensions import db
+from app.extensions.api import Namespace
+from app.extensions.api.parameters import PaginationParameters
+from app.modules.users import permissions
+from app.modules.users.permissions.types import AccessOperation
+
+
+from . import parameters, schemas
+from .models import FileUpload
+
+
+log = logging.getLogger(__name__)  # pylint: disable=invalid-name
+api = Namespace('fileuploads', description='FileUploads')  # pylint: disable=invalid-name
+
+
+@api.route('/')
+@api.login_required(oauth_scopes=['fileuploads:read'])
+class FileUploads(Resource):
+    """
+    Manipulations with FileUploads.
+    """
+
+    @api.permission_required(
+        permissions.ModuleAccessPermission,
+        kwargs_on_request=lambda kwargs: {
+            'module': FileUpload,
+            'action': AccessOperation.READ,
+        },
+    )
+    @api.parameters(PaginationParameters())
+    @api.response(schemas.BaseFileUploadSchema(many=True))
+    def get(self, args):
+        """
+        List of FileUpload.
+
+        Returns a list of FileUpload starting from ``offset`` limited by ``limit``
+        parameter.
+        """
+        return FileUpload.query.offset(args['offset']).limit(args['limit'])
```

Comment:
This would return all uploaded files for all users. I suspect this isn't what we want

Reply:
yup; see above.  this was all very quick cut-n-paste.  mostly to see if my "sketch" of FileUpload is what jp intended in the first place.

Reply:
scrubbed resources.py of many endpoints.  obsolete.